onnxruntime/core/framework/tensorprotoutils.cc

```
@@ -337,6 +342,15 @@ void ConvertRawDataInTensorProto(TensorProto* tensor) {
     num_elements = (tensor->raw_data().size()) / element_size;
     bytes = const_cast<char*>(tensor->mutable_raw_data()->c_str());
   }
+
+  if (ext_data_buf) {
+    if (ext_data_len % element_size != 0) {
+      return;
```

Instead of ignoring error, maybe change to `ORT_ENFORCE(ext_data_len % element_size == 0);`

---

Updated the code using ORT_ENFORCE.